server-common/src/test/java/org/apache/kafka/server/util/timer/MockTimer.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.util.timer;
+
+import org.apache.kafka.server.util.MockTime;
+
+import java.util.Comparator;
+import java.util.PriorityQueue;
+
+public class MockTimer implements Timer {
+    private final PriorityQueue<TimerTaskEntry> taskQueue = new PriorityQueue<>(
+        Comparator.comparingLong(entry -> entry.expirationMs)
+    );
+
+    public final MockTime time;
```

Comment:
should this be private? If it is public for testing, please add a comment. `// visible for testing`

Reply:
I can actually make it private and add an accessor to retrieve it. Regarding adding `// visible for testing`, it makes little sense in this context because the entire class is used for testing. Let me know if the accessor solution works for you.